gmso/abc/abstract_site.py

```diff
@@ -20,8 +19,117 @@
 from gmso.exceptions import GMSOError
 
 PositionType = Union[Sequence[float], np.ndarray, u.unyt_array]
-MoleculeType = NamedTuple("Molecule", name=StrictStr, number=StrictInt)
-ResidueType = NamedTuple("Residue", name=StrictStr, number=StrictInt)
+
+
+class Molecule(GMSOBase):
+    __iterable_attributes__: ClassVar[set] = {
+        "name",
+        "number",
+        "isrigid",
+    }
+
+    __base_doc__: ClassVar[str] = "Molecule label for interaction sites."
+
+    name_: str = Field(
+        "",
+        validate_default=True,
+        description="Name of the molecule",
+        alias="name",
+    )
+    number_: int = Field(
+        0,
+        description="The index/number of the molecule",
+        alias="number",
+    )
+    isrigid_: bool = Field(
+        False,
+        description="Indicate whether the molecule is rigid",
+    )
+    model_config = ConfigDict(
+        alias_to_fields={
+            "name": "name_",
+            "number": "number_",
+            "isrigid": "isrigid_",
+        }
+    )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the molecule."""
+        return self.__dict__.get("name_")
+
+    @property
+    def number(self) -> int:
+        """Return the index/number of the moleucle."""
+        return self.__dict__.get("number_")
+
+    @property
+    def isrigid(self) -> bool:
+        """Return the rigid label of the molecule."""
+        return self.__dict__.get("isrigid_")
+
+    def __hash__(self):
+        return hash(tuple([(name, val) for name, val in self.__dict__.items()]))
+
+    def __eq__(self, other):
+        """Test if two objects are equivalent."""
+        if isinstance(other, (list, tuple)):
+            return all(
+                [val1 == val2 for val1, val2 in zip(self.__dict__.values(), other)]
+            )
+        else:
+            return self.__dict__ == other.__dict__
+
+
+class Residue(GMSOBase):
+    __iterable_attributes__: ClassVar[set] = {
+        "name",
+        "number",
+        "isrigid",
```

Comment:
no isrigid attribute for Residue right now.

Reply:
Fixed